File: TopKFreq.java
//Time complexity: O(n+logk)
//Space complexity: O(n)
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> map=new HashMap<>();
        int[] result=new int[k];
        for(int i=0;i<nums.length;i++)
        {
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
        }
        PriorityQueue<Integer> q=new PriorityQueue<>((a,b)->map.get(a)-map.get(b));
        for(int key:map.keySet())
        {
            q.add(key);
            if(q.size()>k)
            {
                q.poll();
            }
        }
        for(int i=k-1;i>=0;i--)
        {
            result[i]=q.poll();
        }
     return result;   
    }
}

File: topKFreq.py
#Time: O(n)
#Space: O(n)
#Program ran on leetcode successfully

import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        countMap = dict()
        for num in nums:
            countMap[num] = countMap.get(num, 0) + 1
        
        queue = []
        for key in countMap.keys():
            print(countMap[key])
            temp = [countMap[key], key]
            heapq.heappush(queue, (countMap[key], key))
            if (len(queue) > k):
                heapq.heappop(queue)
            
        result = [0] * k
        i = 0
        while (queue):
            result[i] = heapq.heappop(queue)
            i+= 1
        
        return result

        


        